### PR TITLE
Hello Rails React PR

### PR DESCRIPTION
**Pull request by tomdan-ai**
In This Milestone the following were implemented:

- Use webpack to set up frontend react
- Use rails to create database and API
- Use react to fetch API created in rails

### CODE REVIEW

**Comment by addod19** [comment]
## Hi @tomdan-ai, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that is invalid because the README file is not professional

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: README.md [hunk 1]
```diff
@@ -1,24 +1,155 @@
...
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+## :pray: Acknowledgements <a name="acknowledgements"></a>
+
+Thanks to Microverse for their support.
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+  
+## :memo: License <a name="license"></a>
+This project is [MIT](../MIT) licensed.
```
Comment (mikethreels) [inline README.md]: - [ ] the link to the MIT license is not working as expected please fix it before requesting another review.

File: Gemfile [hunk 2]
```diff
@@ -1,31 +1,33 @@
...
+gem 'pg'
 
 # Use the Puma web server [https://github.com/puma/puma]
-gem "puma", "~> 5.0"
+gem 'puma', '~> 5.0'
 
-# Bundle and transpile JavaScript [https://github.com/rails/jsbundling-rails]
-gem "jsbundling-rails"
+# Use JavaScript with ESM import maps [https://github.com/rails/importmap-rails]
+gem 'importmap-rails'
```
Comment (mikethreels) [inline Gemfile]: - [ ] it looks like you removed the required gem "gem "jsbundling-rails"" please make sure to use it in this project as it is a requirement.

File: package.json [hunk 3]
```diff
@@ -1,10 +1,28 @@
...
+    "axios": "^0.25.0",
+    "babel-plugin-transform-react-remove-prop-types": "^0.4.24",
+    "babel-polyfill": "^6.26.0",
+    "prop-types": "^15.8.1",
+    "react": "^18.1.0",
+    "react_ujs": "^2.6.2",
+    "react-dom": "^18.1.0",
+    "react-redux": "^8.0.1",
+    "react-router-dom": "^6.3.0",
+    "redux": "^4.2.0",
```
Comment (mikethreels) [inline package.json]: - [ ] the required dependency "babel-loader" is not installed please make sure to do so.

File: babel.config.js [hunk 4]
```diff
@@ -1,6 +1,99 @@
...
+      isTestEnv && [
+        '@babel/preset-env',
+        {
+          targets: {
+            node: 'current'
+          },
+          modules: 'commonjs'
+        },
+        '@babel/preset-react'
+      ],
```
Comment (mikethreels) [inline babel.config.js]: - [ ] please have a look at [this totorial](https://betterprogramming.pub/react-with-rails-2022-bd28e1fcd355#ad0a) the babel.config.js file should look a lot different and the webpack.config.js file is missing, please make sure to implement these correctly

File: app/views/static/index.html.erb [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<%= javascript_pack_tag 'application' %>
+
+<%= react_component("App") %>
```
Comment (mikethreels) [inline app/views/static/index.html.erb]: - [ ] the view should only have a div with a class called root, the rest of the application should be rendered by react.

File: app/javascript/packs/application.js [hunk 6]
```diff
@@ -16,3 +16,7 @@
 // const imagePath = (name) => images(name, true)
 
 console.log('Hello World from Webpacker')
```
Comment (mikethreels) [inline app/javascript/packs/application.js]: - [ ] please avoid committing any console.log they make your code slower and harder to read.

File: app/javascript/packs/application.js [hunk 7]
```diff
@@ -16,3 +16,7 @@
 // const imagePath = (name) => images(name, true)
```
Comment (mikethreels) [inline app/javascript/packs/application.js]: - [ ] please don't push any commented-out code they will make your codebase look unorganised and harder to read.

File: app/javascript/components/App.js [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+          path="/greeting"
+          element={(
+            <Greeting />
+        )}
+        />
+    </Routes>
+    </BrowserRouter>
+    </Provider>
+  )
+}
```
Comment (mikethreels) [inline app/javascript/components/App.js]: - [ ] everything should exist in the application.js file not in your app.js file.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Tom, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job setting up the project.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @tomdan-ai,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Well organized and clean project ✔️
- Professional readme file ✔️
- No linter errors ✔️

thanks-awesome

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**